internal/rootcoord/root_coord.go

```diff
@@ -855,24 +858,34 @@ func (c *Core) Register() error {
 
 func (c *Core) Init() error {
 	var initError error = nil
+	if c.kvBaseCreate == nil {
+		c.kvBaseCreate = func(root string) (kv.TxnKV, error) {
+			return etcdkv.NewEtcdKV(Params.EtcdEndpoints, root)
+		}
+	}
 	c.initOnce.Do(func() {
 		connectEtcdFn := func() error {
 			if c.etcdCli, initError = clientv3.New(clientv3.Config{Endpoints: Params.EtcdEndpoints, DialTimeout: 5 * time.Second}); initError != nil {
 				log.Error("RootCoord, Failed to new Etcd client", zap.Any("reason", initError))
 				return initError
 			}
-			var ms *metaSnapshot
-			ms, initError = newMetaSnapshot(c.etcdCli, Params.MetaRootPath, TimestampPrefix, 1024)
+			if c.kvBase, initError = c.kvBaseCreate(Params.KvRootPath); initError != nil {
```

Comment:
what is the reason why we need. kvBaseCreate function? 
why not just etcdkv.NewEtcdKV?

Reply:
`kvBaseCreate` can provide injection point for changing `TxnKV` implementation, which provided a method to inject failure in ut.